### ISSUE TITLE
Output Node Name?

### ISSUE THREAD

**Issue by AdamWP** [TensorBox/TensorBox#180]
I'm trying to combine a checkpoint with the input graph using freeze_graph.py.  However freeze_graphy.py needs the input graph and the output_node_names.  

I believe I managed to save the input_graph (or at least I assume it's the right 8.23 MB file)  by adding the following to line 447 of train.py.  

input_graph_name = "input_graph.pb"
input_graph_dir = "models"
tf.train.write_graph(sess.graph_def, input_graph_dir, input_graph_name) 

However I'm having issues finding the output_node_names.  

Is there a way to do this?  Or does anyone know what it or they are?

**Comment by mkreich** [comment]
I did guess the names of the nodes by looking at the Tensorboard Graph and the pbtxt file.
The names were 'add' for the rectangles and 'Reshape_2'  for the confidence values. (anything after softmax  is a good candidate, I think)
Since frezzing/defreezing produces the same values as building forward, this seems to be the right call (I hope) :)

**Comment by AdamWP** [comment]
It looks like 'add' isn't in the graph.  I get an error when I run  "python freeze_graph.py --input_graph=input_graph.pb --input_checkpoint=save.ckpt --output_graph=frozen_graph.pb   --output_node_names=add,Reshape_2 "

**Comment by PumayHui** [comment]
I get more than 1000 output_node_names by this : 
`bazel-bin/tensorflow/tools/graph_transforms/summarize_graph --in_graph=model.pbtxt`
How can I know which node is the final output? Anyone do this? Thanks so much!

**Comment by rickragv** [comment]
has anyone found solution

**Comment by PumayHui** [comment]
@mkreich  I follow your advice. But the error is ：
add is not in graph.
